src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/ViewExecutor.cs

```diff
@@ -257,5 +259,17 @@ public class ViewExecutor
                 await writer.FlushAsync();
             }
         }
+
+        protected virtual void OnExecuting(ViewContext viewContext)
```

Comment:
What do you think about using the same pattern we have with `ITempDataCallback`? The name needs work, but the pattern is pretty straight forward.